spec/compiler/codegen/is_nil_spec.cr

```diff
@@ -0,0 +1,27 @@
+require "../../spec_helper"
+
+describe "Codegen: nil?" do
+  it "codegens .nil? false for simple type" do
+    run("1.nil?").to_b.should be_false
+  end
+
+  it "codegens .nil? true for nil" do
+    run("nil.nil?").to_b.should be_true
+  end
+
+  it "codegens .nil?() false for simple type" do
```

Comment:
We don't need duplicate codegen specs for syntax alternatives. There should be specs that ensure both method calls (with and without parentheses) are semantically equal. By logical conclusion, codegen must also be identical so it doesn't need separate specs.
Ditto all below.